### PR TITLE
ref(codeowners): Move handle_owner_assignment to new task

### PR DESCRIPTION
**Pull request by jangjodi**
Move get_issue_owers method, handle_group_owners from handle_owner_assignment
and handle_auto_assignment to a new task.

WOR-2716

### CODE REVIEW

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -220,39 +219,47 @@ def handle_owner_assignment(job):
...
-                        issue_owners = []
-                    else:
 
-                        issue_owners = ProjectOwnership.get_issue_owners(project.id, event.data)
+@instrumented_task(
+    name="sentry.tasks.post_process_handle_owner_assignment",
+    queue="post_process_handle_owner_assignment",
+)
+def process_owner_assignments(
+    project: Project, event: GroupEvent | Event, group: Group, job: any, issue_owners_key: str
```
[the remark this answers is not in the document]
Comment (jangjodi) [reply src/sentry/tasks/post_process.py]: thanks! I changed these

**Review by NisanthanNanthakumar — COMMENTED** [review]
Looks good!

**Review by NisanthanNanthakumar — APPROVED** [review]

**Review by wedamija — COMMENTED** [review]
Since you're adding a new queue here you'll need to make a ticket for ops to assign workers to it

**Comment by jangjodi** [comment]
> Since you're adding a new queue here you'll need to make a ticket for ops to assign workers to it

Thank you @wedamija ! Would I wait to merge this PR after the workers have been assigned?

**Pull request merged**